src/avro/src/decode.rs

```diff
@@ -86,51 +88,59 @@ fn build_ts_value(value: i64, unit: TsUnit) -> Result<Value, Error> {
     ))
 }
 
+/// A convenience trait for types that are both readable and skippable.
+///
+/// A blanket implementation is provided for all types that implement both
+/// [`Read`] and [`Skip`].
+pub trait AvroRead: Read + Skip {}
+
+impl<T> AvroRead for T where T: Read + Skip {}
+
+/// A trait that allows for efficient skipping forward while reading data.
 pub trait Skip {
-    fn skip(&mut self, len: usize) -> Result<(), Error>;
+    /// Advance the cursor by `len` bytes.
+    ///
+    /// If possible, the implementation should be more efficient than calling
+    /// [`Read::read`] and discarding the resulting bytes.
+    ///
+    /// Calling `skip` with a `len` that advances the cursor past the end of the
+    /// underlying data source is permissible. The only requirement is that the
+    /// next call to [`Read::read`] indicates EOF.
+    ///
+    /// # Errors
+    ///
+    /// Can return an error in all the same cases that [`Read::read`] can.
+    fn skip(&mut self, len: usize) -> Result<(), io::Error>;
 }
 
-impl Skip for std::fs::File {
-    fn skip(&mut self, len: usize) -> Result<(), Error> {
+impl Skip for File {
+    fn skip(&mut self, len: usize) -> Result<(), io::Error> {
         self.seek(SeekFrom::Current(len as i64))?;
         Ok(())
     }
 }
 
 impl Skip for &[u8] {
-    fn skip(&mut self, len: usize) -> Result<(), Error> {
-        if len > self.len() {
-            bail!(
-                "Failed to skip {} bytes in buffer: length is {}.",
-                len,
-                self.len()
-            );
-        }
+    fn skip(&mut self, len: usize) -> Result<(), io::Error> {
```

Comment:
Erroring here is intentional. If you’re trying to read an avro buffer which says the next 10 bytes are a string but there are only 5 bytes left in your buffer, there’s no chance to salvage any correct behavior. Either your buffer is corrupt or there is a bug in mz-avro.

Actually the bug here IMO is that the other Skip implementations don’t do this bounds check.